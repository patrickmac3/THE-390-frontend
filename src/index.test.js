import React from 'react';
import ReactDOM from 'react-dom/client';
import { AuthProvider } from './utils/hooks/AuthContext';
import { PropertyProvider } from './utils/hooks/PropertyContext';
import { ProfileProvider } from './utils/hooks/ProfileContext';
import App from './App';
import { act } from 'react-dom/test-utils';
import TestRenderer from 'react-test-renderer'; // ES6

// Mocking the CSS imports as Jest cannot parse CSS.
jest.mock('./index.css', () => {});
jest.mock('./bootstrap.min-3.css', () => {});

describe('Index Component Rendering', () => {
  let container = null;

  beforeEach(() => {
    // Setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // Cleanup on exiting
    document.body.removeChild(container);
    container = null;
  });

  it('renders without crashing', () => {
    act(() => {
      const root = ReactDOM.createRoot(container);
      root.render(
        <React.StrictMode>
          <AuthProvider>
            <PropertyProvider>
              <ProfileProvider>
                <App />
              </ProfileProvider>
            </PropertyProvider>
          </AuthProvider>
        </React.StrictMode>
      );
    });
  });

  it('renders App and checks for structured tree', () => {
    const testRenderer = TestRenderer.create(
      <React.StrictMode>
        <AuthProvider>
          <PropertyProvider>
            <ProfileProvider>
              <App />
            </ProfileProvider>
          </PropertyProvider>
        </AuthProvider>
      </React.StrictMode>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot(); // This will create a snapshot of your component hierarchy
  });
});
