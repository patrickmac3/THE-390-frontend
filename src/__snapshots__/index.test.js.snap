// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Component Rendering renders App and checks for structured tree 1`] = `
Array [
  " ",
  " ",
  <header>
    <nav
      className="bg-body-tertiary navbar navbar-expand-sm navbar-light"
    >
      <div
        className="container"
      >
        <a
          className="active navbar-brand"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          CondoCare
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            className="me-auto navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="nav-link"
              data-rr-ui-event-key="/login"
              disabled={false}
              href="/login"
              onClick={[Function]}
              style={null}
            >
              <i
                className="fas fa-user"
              />
               LOGIN
            </a>
          </div>
          <div
            className="ms-auto justify-content-end navbar-nav nav-tabs"
            onKeyDown={[Function]}
          />
        </div>
      </div>
    </nav>
  </header>,
  <main
    className="py-3"
  >
    <div
      className="container"
    >
      <div
        className="container"
      >
        <div
          className="row justify-content-center"
        >
          <div
            className="col-lg-10 col-md-13 mt-5"
          >
            <div
              className="d-flex flex-column align-items-center"
            >
              <h1
                style={
                  Object {
                    "fontSize": "50px",
                    "fontWeight": "bold",
                    "marginBottom": "30px",
                  }
                }
              >
                Welcome to CondoCare!
              </h1>
              <h4
                style={
                  Object {
                    "marginBottom": "30px",
                  }
                }
              >
                Get Ready for Easy Property Management!
              </h4>
            </div>
            <div
              className="carousel slide carousel-fade"
              onKeyDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="carousel-indicators"
              >
                <button
                  aria-current={true}
                  aria-label="Slide 1"
                  className="active"
                  data-bs-target=""
                  onClick={[Function]}
                  type="button"
                />
                <button
                  aria-current={false}
                  aria-label="Slide 2"
                  data-bs-target=""
                  onClick={[Function]}
                  type="button"
                />
                <button
                  aria-current={false}
                  aria-label="Slide 3"
                  data-bs-target=""
                  onClick={[Function]}
                  type="button"
                />
              </div>
              <div
                className="carousel-inner"
              >
                <div
                  className="active carousel-item"
                >
                  <div
                    className="carousel-image-container"
                    style={
                      Object {
                        "overflow": "hidden",
                        "paddingTop": "56.25%",
                        "width": "100%",
                      }
                    }
                  >
                    <img
                      alt="First slide"
                      src="property.jpeg"
                      style={
                        Object {
                          "height": "100%",
                          "left": "50%",
                          "objectFit": "cover",
                          "position": "absolute",
                          "top": "50%",
                          "transform": "translate(-50%, -50%)",
                          "width": "100%",
                        }
                      }
                    />
                  </div>
                  <div
                    className="carousel-caption"
                  >
                    <h3>
                      First slide label
                    </h3>
                    <p>
                      Nulla vitae elit libero, a pharetra augue mollis interdum.
                    </p>
                  </div>
                </div>
                <div
                  className="carousel-item"
                >
                  <div
                    className="carousel-image-container"
                    style={
                      Object {
                        "overflow": "hidden",
                        "paddingTop": "56.25%",
                        "width": "100%",
                      }
                    }
                  >
                    <img
                      alt="First slide"
                      src="property1.png"
                      style={
                        Object {
                          "height": "100%",
                          "left": "50%",
                          "objectFit": "cover",
                          "position": "absolute",
                          "top": "50%",
                          "transform": "translate(-50%, -50%)",
                          "width": "100%",
                        }
                      }
                    />
                  </div>
                  <div
                    className="carousel-caption"
                  >
                    <h3>
                      Second slide label
                    </h3>
                    <p>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    </p>
                  </div>
                </div>
                <div
                  className="carousel-item"
                >
                  <div
                    className="carousel-image-container"
                    style={
                      Object {
                        "overflow": "hidden",
                        "paddingTop": "56.25%",
                        "width": "100%",
                      }
                    }
                  >
                    <img
                      alt="First slide"
                      src="property2.jpeg"
                      style={
                        Object {
                          "height": "100%",
                          "left": "50%",
                          "objectFit": "cover",
                          "position": "absolute",
                          "top": "50%",
                          "transform": "translate(-50%, -50%)",
                          "width": "100%",
                        }
                      }
                    />
                  </div>
                  <div
                    className="carousel-caption"
                  >
                    <h3>
                      Third slide label
                    </h3>
                    <p>
                      Praesent commodo cursus magna, vel scelerisque nisl consectetur.
                    </p>
                  </div>
                </div>
              </div>
              <a
                className="carousel-control-prev"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  aria-hidden="true"
                  className="carousel-control-prev-icon"
                />
                <span
                  className="visually-hidden"
                >
                  Previous
                </span>
              </a>
              <a
                className="carousel-control-next"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  aria-hidden="true"
                  className="carousel-control-next-icon"
                />
                <span
                  className="visually-hidden"
                >
                  Next
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>,
  <footer>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="text-center py-3 col"
        >
          Copyright Â© CondoCare
        </div>
      </div>
    </div>
  </footer>,
  " ",
  " ",
]
`;
